src/Extension.ml

```diff
@@ -15,9 +15,19 @@ module Client = struct
 end
 
 module Server = struct
+  let refmtSettings =
+    Settings.create ~scope:Workspace ~key:"refmt.width" ~ofJson:Json.Decode.int
+      ~toJson:Json.Encode.int
+
   let make (toolchain : Toolchain.resources) :
       Vscode.LanguageClient.serverOptions =
     let command, args = Toolchain.getLspCommand toolchain in
+    let args =
+      match Settings.get refmtSettings with
+      | None -> args
+      | Some refmtWidth ->
+        Array.concat [ args; [| "--refmt-width"; string_of_int refmtWidth |] ]
```

Comment:
A command line option is not the correct place to pass this option. Preferably:

* It should be passed as a custom option to the formatting request (see the LSP spec).
* If this isn't possible, there's a mechanism for setting workspace configuration options.

Reply:
@rgrinberg I'm not sure about adding options to the requests from the client. We do have an option to send `initializationOptions` when we start the server

https://github.com/microsoft/vscode-languageserver-node/blob/master/client/src/client.ts#L508